src/main/java/com/arpnetworking/metrics/impl/BaseFileSink.java

```diff
@@ -0,0 +1,374 @@
+import ch.qos.logback.core.rolling.RollingFileAppender;
+import ch.qos.logback.core.rolling.TimeBasedRollingPolicy;
+import com.arpnetworking.metrics.Sink;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Base class for <code>Sink</code> implementations which write to files. This
+ * implementation uses Logback as the underlying implementation to write events
+ * to disk. It is designed not to interfere with Logback or SLF4J usage for
+ * application logging.
+ *
+ * @author Ville Koskela (vkoskela at groupon dot com)
+ */
+/* package private */ abstract class BaseFileSink implements Sink {
+
+    /**
+     * Accessor for the <code>Logger</code> instance to write to.
+     *
+     * @return The <code>Logger</code> instance to write to.
+     */
+    protected Logger getMetricsLogger() {
+        return _metricsLogger;
+    }
+
+    private TimeBasedRollingPolicy<ILoggingEvent> createRollingPolicy(
+            final String extension,
+            final String fileNameWithoutExtension,
+            final int maxHistory,
+            final boolean compress) {
+
+        // TODO(vkoskela): Use LogbackSteno's random compression strategy [MAI-413]
+        final TimeBasedRollingPolicy<ILoggingEvent> rollingPolicy = new TimeBasedRollingPolicy<>();
+        rollingPolicy.setContext(_loggerContext);
+        rollingPolicy.setMaxHistory(maxHistory);
+        rollingPolicy.setCleanHistoryOnStart(true);
+        if (compress) {
+            rollingPolicy.setFileNamePattern(fileNameWithoutExtension + DATE_EXTENSION + extension + GZIP_EXTENSION);
+        } else {
+            rollingPolicy.setFileNamePattern(fileNameWithoutExtension + DATE_EXTENSION + extension);
+        }
+
+        return rollingPolicy;
+    }
+
+    private FileAppender<ILoggingEvent> createRollingAppender(
+            final String fileName,
+            final boolean prudent,
+            final TimeBasedRollingPolicy<ILoggingEvent> rollingPolicy,
+            final Encoder<ILoggingEvent> encoder) {
+        final RollingFileAppender<ILoggingEvent> rollingAppender = new RollingFileAppender<>();
+        rollingAppender.setContext(_loggerContext);
+        rollingAppender.setName("query-log");
+        rollingAppender.setFile(fileName);
+        // TODO(vkoskela): Implement prudent mode [MAI-415]
+        //rollingAppender.setPrudent(prudent);
+        rollingAppender.setAppend(true);
+        rollingAppender.setRollingPolicy(rollingPolicy);
+        rollingAppender.setEncoder(encoder);
+        return rollingAppender;
+    }
+
+    private Appender<ILoggingEvent> createAsyncAppender(final Appender<ILoggingEvent> appender) {
+        final AsyncAppender asyncAppender = new AsyncAppender();
+        asyncAppender.setContext(_loggerContext);
+        asyncAppender.setDiscardingThreshold(0);
+        asyncAppender.setName("query-log-async");
+        asyncAppender.setQueueSize(500);
```

Comment:
We could, none of these construction methods are because they refer to _loggerContext which is non-static. We could pass that around of course. What's your preference?